src/cli/serve/__tests__/reload_logging_config.js

```diff
@@ -0,0 +1,64 @@
+import { spawn } from 'child_process';
+import { promisify } from 'bluebird';
+import { writeFile, readFile } from 'fs';
+import { relative, resolve } from 'path';
+import { safeDump } from 'js-yaml';
+import mkdirp from 'mkdirp';
+import rimraf from 'rimraf';
+import readYamlConfig from '../read_yaml_config';
+import expect from 'expect.js';
+const testConfigFile = follow(`fixtures/reload_logging_config/kibana.test.yml`);
+const pwriteFile = promisify(writeFile);
+const preadFile = promisify(readFile);
+const mkdir = promisify(mkdirp);
+const primraf = promisify(rimraf);
+
+function follow(file) {
+  return relative(process.cwd(), resolve(__dirname, file));
+}
+
+async function setLoggingDest(to) {
+  const conf = readYamlConfig(testConfigFile);
+  conf.logging = conf.logging || {};
+  conf.logging.dest = follow(to);
+  const yaml = safeDump(conf);
+  await pwriteFile(testConfigFile, yaml);
+  return conf;
+}
+
+function wait(time) {
+  return new Promise(resolve => setTimeout(() => resolve(), time));
+}
+
+describe(`Server logging configuration`, function () {
+  it(`should wait`, async function () {
+    const testLogsDirectory = follow(`logs`);
+    await primraf(testLogsDirectory);
+    await mkdir(testLogsDirectory);
+
+    const conf = await setLoggingDest(`logs/first.log`);
+    const child = spawn(`./bin/kibana`, [`--config`, testConfigFile]);
+
+    // wait for just a bit
+    await wait(5000);
```

Comment:
eek! I'm thinking we could do a logging format change, rather than a location change, and then parse the json output of the process to know when it's change?

 1. before `SIGHUP`:
    ```yaml
    logging: 
      json: true
    ```
 1. wait for `{ ... "listing" ... }` log line
 1. update yaml file: `logging.json: false`
 1. wait for logging to show up with `[tag]` and such.